### DISCUSSION TITLE
Query has wrong type

### DISCUSSION THREAD

**Discussion by ilsnk** [dotnet/WatsonWebserver#97] [General]
Quary can has recurring keys, correct use NameValueCollection 
![image](https://user-images.githubusercontent.com/44324620/175957731-3e95eece-86f3-4ca1-80d3-ca74fac519f2.png)

**Comment by ilsnk** [comment]
and why WatsonWebserver not supported Basic Authentication ? (like on screen)
![image](https://user-images.githubusercontent.com/44324620/175962002-985e9d1c-02a9-4bc6-a5bc-9e2521c875ed.png)

**Comment by jchristn** [comment]
Authentication should be a function you implement in your ```PreRequestRouting``` handler or elsewhere.  It was never intended to  have the library manage authentication for you.

Regarding use of NameValueCollection vs Dictionary, the server follows the RFC in that multiple values provided with multiple keys of the same name can be represented as a single value separated by commas.  This SO answer may be helpful: https://stackoverflow.com/questions/3096888/standard-for-adding-multiple-values-of-a-single-http-header-to-a-request-or-resp

Cheers, Joel

**Reply by ilsnk** [reply]
how then to call the default authorization window, for example, as when connecting to the wifi settings or what I threw (above screen)

**Reply by jchristn** [reply]
Try the first answer [here](https://stackoverflow.com/questions/61204987/how-to-make-the-browser-prompt-for-default-user-authentication-pop-up) and let me know.

**Reply by jchristn** [reply]
Essentially when you receive a request without credentials, return a status 401 and set an HTTP header according to what you see in that link.  That should prompt the browser to then prompt the user for credentials to attach to the request.
